torch/distributed/_tensor/api.py

```diff
@@ -649,6 +650,69 @@ def distribute_tensor(
         # 1. if the we can further shard this DTensor if the two device mesh belong to
         #   the same parenet mesh and further sharding is possible.
         # 2. check if device mesh and placements are the same
+
+        # use case: right-to-left tensor distribution i.e. nested distribute_tensor()
+        # over 1-d meshes for composable parallelisms. This branch specifically allows
+        # for FSDP2 sharding where the tensor is first sharded on a 1-d device mesh and
+        # then passed to another distribute_tensor() call along with another 1-d device
+        # mesh, and potentially so on.
+        input_parent_mesh = _mesh_resources.get_parent_mesh(device_mesh)
+        dtensor_parent_mesh = _mesh_resources.get_parent_mesh(tensor.device_mesh)
+        right_to_left_distribute = (
```

Comment:
We don't actually check if we are sharding from right to left. If the input mesh is actually in the right of the tensor's submesh, we should raise an exception.